ksqldb-execution/src/main/java/io/confluent/ksql/execution/ddl/commands/AlterSourceCommand.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.ddl.commands;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.confluent.ksql.name.SourceName;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+
+public class AlterSourceCommand implements DdlCommand {
+  private final SourceName sourceName;
+  private final String ksqlType;
+  private final LogicalSchema newColumns;
+
+
+  public AlterSourceCommand(
+      @JsonProperty(value = "sourceName", required = true) final SourceName sourceName,
+      @JsonProperty(value = "ksqlType", required = true) final String ksqlType,
+      @JsonProperty(value = "newColumns", required = true) final LogicalSchema newColumns
```

Comment:
passing in a `LogicalSchema` to represent the new columns seems... odd? we should probably be passing down just the new columns - users shouldn't be, for example, creating new `KEY` columns (at the moment at least), and the new columns shouldn't include any of the psuedocolumns.

we can see this in the way that you use it (you only get the columns):
```java
alterSource.getNewColumns().columns()
```

Reply:
This was done because the Column class could not be deserialized:

```
ERROR Did not write the command to the command topic as it could not be deserialized. This is a bug! Please raise a Github issue containing the series of commands you ran to get to this point.
Cannot construct instance of `io.confluent.ksql.schema.ksql.Column` (no Creators, like default construct, exist): cannot deserialize from Object value (no delegate- or property-based Creator)
```

But yeah, using LogicalSchema here is pretty odd. Perhaps creating a new class for this purpose would make more sense.

Reply:
That makes sense, I think all you'd need to do is add a serde module. You can see `LogicalSchemaDeserializer` and `LogicalSchemaSerializer` (and they're hooked up in `KsqlTypesDeserializationModule`)